### PR TITLE
chore: update code_health third parties action to their shas

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes
Update the code health third parties action to their sha

### CODE REVIEW

**Review by gssbzn — CHANGES_REQUESTED** [review]
not the same binary but the same rules, does the binary matter?

**Comment by gssbzn** [comment]
If you see the code they use it as a library, https://github.com/golangci/golangci-lint/blob/8f348db7bbd6366c528a369ad0b02505afd265c8/pkg/golinters/staticcheck/staticcheck.go#L4C22-L4C33 the note is just because of beef between the two projects 

https://github.com/golangci/golangci-lint/pull/2894

**Comment by andreaangiolillo** [comment]
ah okay, I was tricked by `It's not the same thing as the staticcheck binary.`. I will reuse this PR to update the action to ping third parties to sha then

File: .github/workflows/code-health.yml [hunk 1]
```diff
@@ -10,17 +10,17 @@ jobs:
     name: lint
     runs-on: ubuntu-latest
     steps:
-      - uses: GitHubSecurityLab/actions-permissions/monitor@v1
+      - uses: GitHubSecurityLab/actions-permissions/monitor@e83639073897b68de235d81a783b2221cc13def7
         with:
           config: ${{ vars.PERMISSIONS_CONFIG }}
-      - uses: actions/checkout@v4
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332
```
Comment (gssbzn) [inline .github/workflows/code-health.yml:16]: my understanding was that actions supported by GH could remain on numbers and not shas, has this changed?

Comment (gssbzn) [reply .github/workflows/code-health.yml:16]: using shas is actually really bad for security scanning tools as you may be able to  see from https://github.com/mongodb/mongodb-atlas-cli/pull/3063

Comment (andreaangiolillo) [reply .github/workflows/code-health.yml:16]: I was unaware of this, could you share where it was decided?

Comment (andreaangiolillo) [reply .github/workflows/code-health.yml:16]: no problem if you don't find it. It makes sense that GH actions can be trusted

**Pull request closed without merging**